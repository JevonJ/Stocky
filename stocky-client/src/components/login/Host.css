.loginHost-cover {
  padding-left: 30%;
  padding-right: 30%;
}
.loginHost-cover .btn-lg {
  padding: .75rem 1.25rem;
  margin: 1em;
  font-weight: 700;
}
.loginHost-cover p {
  margin: 1em;  
}