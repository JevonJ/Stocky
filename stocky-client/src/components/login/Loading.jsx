import React from 'react';
import ReactLoading from 'react-loading';

const Loading = () => (
  <div style={{ paddingLeft: '46%' }}>
    <ReactLoading type="bars" color="#fff" width={100} height={100} />
  </div>
);

export default Loading;
