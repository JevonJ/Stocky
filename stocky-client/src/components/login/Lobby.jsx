import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Button, Fade, Row, Col, Container, Table } from 'reactstrap';
import CountDown from 'react-countdown-clock';
import PropTypes from 'prop-types';

class Lobby extends Component {
  static renderUsers(player, index) {
    return (
      <tr key={player}>
        <th scope="row">{index + 1}</th>
        <td>{player.name}</td>
      </tr>
    );
  }

  componentWillMount() {
    const { user, history } = this.props;
    if (user.constructor === Object && Object.keys(user).length === 0) {
      history.replace('/');
    }
  }

  onStart() {
    const { user } = this.props;
    this.props.socket.emit('start_game', user.room);
  }

  startGame() {
    const { history } = this.props;
    history.push('/simulator');
  }

  render() {
    const { user, players, time } = this.props;
    return (
      <Fade in tag="div" timeout={200}>
        {/* <NavigationPrompt
          beforeConfirm={this.cleanup}
          renderIfNotActive
          disableNative
          when={(crntLocation, nextLocation) => {
            console.log('xxxxxxxxxxxxxxxxxxxxxxxxxxxx');
            console.log(crntLocation);
            console.log(nextLocation);
            return !nextLocation || !nextLocation.pathname.startsWith(crntLocation.pathname);
          }}
        >
          {({ isActive, onCancel, onConfirm }) => {
            if (isActive) {
              return (
                <Modal isOpen>
                  <ModalHeader toggle={this.toggle}>Modal title</ModalHeader>
                  <ModalFooter>
                    <Button color="primary" onClick={this.toggle}>Do Something</Button>{' '}
                    <Button color="secondary" onClick={this.toggle}>Cancel</Button>
                  </ModalFooter>
                </Modal>
              );
            }
            return (
              <div>This is probably an anti-pattern but ya know...</div>
            );
          }}
        </NavigationPrompt> */}
        <main role="main" className="inner cover">
          <h1 className="cover-heading">Lobby : {user && user.room}</h1>
          <p className="lead">Connected Players</p>
        </main>
        <Container>
          <Row>
            <Col sm="12" md="{{ size: 8, offset: 2 }}">
              <Table hover responsive>
                <thead>
                  <tr>
                    <th>#</th>
                    <th>Username</th>
                  </tr>
                </thead>
                <tbody>
                  {
                    players && players.map((player, index) => Lobby.renderUsers(player, index))
                  }
                </tbody>
              </Table>
            </Col>
          </Row>
          <Row>
            {time.start_time > 0 &&
              <div style={{ width: '100%' }}>
                <Col sm="12">
                Game starting in...
                </Col>
                <div style={{ paddingLeft: '40%', paddingRight: '40%', paddingTop: '10px' }}>
                  <CountDown
                    seconds={time.start_time}
                    color="#fff"
                    alpha={0.9}
                    size={100}
                    onComplete={() => this.startGame()}
                  />
                </div>
              </div>
            }
            {time.start_time === 0 && user.host &&
              <Col sm="12">
                <Button
                  outline
                  color="success"
                  size="lg"
                  onClick={() => this.onStart()}
                >
                    Start
                </Button>
              </Col>
            }
          </Row>
        </Container>
      </Fade>
    );
  }
}

const mapStateToProps = ({ user, players, time }) => ({
  user,
  players,
  time,
});

Lobby.propTypes = {
  user: PropTypes.shape({}).isRequired,
  socket: PropTypes.shape({
    emit: PropTypes.func.isRequired,
  }).isRequired,
  time: PropTypes.shape({}).isRequired,
  players: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  history: PropTypes.shape({}).isRequired,
};

export default connect(mapStateToProps)(Lobby);
