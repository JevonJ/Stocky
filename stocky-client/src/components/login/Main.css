/*
 * Globals
 */

/* Links */
.loginMain-a,
.loginMain-a:focus {
  color: #fff;
}

.loginMain-a:hover {
  color: #aaa;
  text-decoration: none;
}

.loginMain-btn-link {
  color: #fff;
}

.loginMain-btn-link:hover {
  color: #aaa;
  text-decoration: none;
}

.loginMain-container {
  background-color: rgba(0, 0, 0, .8);
}

/*
 * Header
 */
.loginMain-masthead {
  margin-bottom: 2rem;
  padding: 2em 2em 0 2em;
}

.loginMain-masthead-brand {
  margin-bottom: 0;
}

.loginMain-nav-masthead .loginMain-nav-link {
  padding: .25rem 0;
  font-weight: 700;
  color: rgba(255, 255, 255, .5);
  background-color: transparent;
  border-bottom: .25rem solid transparent;
}

.loginMain-nav-masthead .loginMain-nav-link:hover,
.loginMain-nav-masthead .loginMain-nav-link:focus {
  border-bottom-color: rgba(255, 255, 255, .25);
}

loginMain-nav-link {
  text-decoration-line: none;
}

.loginMain-nav-masthead .loginMain-nav-link + .loginMain-nav-link {
  margin-left: 1rem;
}

.loginMain-nav-masthead .active {
  color: #fff;
  border-bottom-color: #fff;
}

@media (min-width: 48em) {
  .loginMain-masthead-brand {
    float: left;
  }
  .loginMain-nav-masthead {
    float: right;
  }
}

/*
 * Footer
 */
.loginMain-mastfoot {
  color: rgba(255, 255, 255, .5);
}