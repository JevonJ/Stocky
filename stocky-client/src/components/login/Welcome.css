/*
 * Cover
 */
.loginWelcome-cover {
  padding: 0 1.9rem;
}
.loginWelcome-cover .btn-lg {
  padding: .75rem 1.25rem;
  margin: 1em;
  font-weight: 700;
}
.loginWelcome-cover p {
  margin: 1em;  
}

