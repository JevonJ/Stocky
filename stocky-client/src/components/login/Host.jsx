import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Button, Fade, Form, FormGroup, Label, Input, FormFeedback, InputGroup, InputGroupAddon, CustomInput } from 'reactstrap';
import PropTypes from 'prop-types';
import FontAwesome from 'react-fontawesome';

import './Host.css';

class Host extends Component {
  constructor() {
    super();

    this.state = {
      hostForm: {
        room: '',
        username: '',
        isPrivate: false,
        password: '',
        rounds: 15,
      },
      visibility: false,
      roomNameError: '',
      userNameError: '',
      passwordError: '',
      roundsError: '',
    };
  }

  onInputChange({ target: { name, value } }) {
    const hostForm = { ...this.state.hostForm };
    hostForm[name] = value.replace(/\s/g, '');

    if (name === 'room') {
      const { rooms } = this.props;
      const roomTaken = !(rooms.indexOf(value) < 0);
      this.setState({
        hostForm,
        roomNameError: roomTaken ? 'That name is already taken..' : '',
      });
    }

    this.setState({ hostForm });
  }

  onCheckboxChange({ target: { name, checked } }) {
    const hostForm = { ...this.state.hostForm };
    hostForm[name] = checked;
    this.setState({ hostForm });
  }

  createGame(e) {
    const { hostForm, roomNameError } = this.state;

    if (hostForm.room.length === 0) {
      this.setState({
        roomNameError: 'Please enter a valid room name.',
      });
      return;
    }

    if (hostForm.username.length === 0) {
      this.setState({
        userNameError: 'Please enter a valid user name.',
      });
      return;
    }

    if (hostForm.rounds.length === 0) {
      this.setState({
        roundsError: 'Please enter number of rounds you want to play',
      });
      return;
    }

    if (hostForm.rounds < 15) {
      this.setState({
        roundsError: 'Rounds should be higher than 15',
      });
      return;
    }

    if (hostForm.isPrivate && hostForm.password.length === 0) {
      this.setState({
        passwordError: 'Please enter a password.',
      });
      return;
    }

    if (roomNameError === '') {
      e.target.setAttribute('disabled', 'disabled');
      this.props.socket.emit('create_game', this.state.hostForm);
    }
  }

  changeVisibility() {
    this.setState({
      visibility: !this.state.visibility,
    });
  }

  render() {
    const {
      hostForm: {
        rounds, room, username, isPrivate, password,
      },
      roomNameError,
      userNameError,
      passwordError,
      roundsError,
    } = this.state;
    const { history } = this.props;
    return (
      <Fade in tag="div" timeout={500}>
        <main role="main" className="inner loginHost-cover">
          <h1 className="loginHost-cover-heading">Host A Game</h1>
          <Form>
            <FormGroup row>
              <Label for="room">Game Room Name</Label>
              <Input
                invalid={roomNameError !== ''}
                valid={room.length > 0 ? !roomNameError : false}
                type="text"
                name="room"
                id="room"
                autoComplete="off"
                onChange={e => this.onInputChange(e)}
                value={room}
              />
              <FormFeedback valid>Sweet! that name is available</FormFeedback>
              <FormFeedback>{roomNameError}</FormFeedback>
            </FormGroup>
            <FormGroup row>
              <Label for="username">User name</Label>
              <Input
                type="text"
                name="username"
                id="username"
                autoComplete="off"
                invalid={userNameError !== ''}
                onChange={e => this.onInputChange(e)}
                value={username}
              />
              <FormFeedback>{userNameError}</FormFeedback>
            </FormGroup>
            <FormGroup row>
              <Label for="rounds">Number of Rounds</Label>
              <Input
                type="number"
                name="rounds"
                id="rounds"
                autoComplete="off"
                invalid={roundsError !== ''}
                max={50}
                min={15}
                onChange={e => this.onInputChange(e)}
                value={rounds}
              />
              <FormFeedback>{roundsError}</FormFeedback>
            </FormGroup>
            <FormGroup row>
              <CustomInput
                type="checkbox"
                name="isPrivate"
                id="isPrivate"
                label="Private Game"
                onChange={e => this.onCheckboxChange(e)}
                value={isPrivate}
              />
            </FormGroup>
            { isPrivate &&
              <FormGroup row>
                <Label for="password">Password</Label>
                <InputGroup>
                  <Input
                    type={this.state.visibility ? 'text' : 'password'}
                    name="password"
                    id="password"
                    value={password}
                    invalid={passwordError !== ''}
                    autoComplete="password"
                    onChange={e => this.onInputChange(e)}
                  />

                  <InputGroupAddon addonType="append">
                    <Button color="secondary" onClick={() => this.changeVisibility()}>
                      <FontAwesome
                        className="super-crazy-colors"
                        name={this.state.visibility ? 'eye' : 'eye-slash'}
                        size="2x"
                      />
                    </Button>
                  </InputGroupAddon>
                </InputGroup>
                <FormFeedback>{passwordError}</FormFeedback>
              </FormGroup>
            }
          </Form>
          <p className="lead">
            <Button
              outline
              color="success"
              size="lg"
              onClick={e => this.createGame(e)}
            >
              Create Game
            </Button>
            <Button
              outline
              color="secondary"
              size="lg"
              onClick={() => history.goBack()}
            >
              Cancel
            </Button>
          </p>
        </main>
      </Fade>
    );
  }
}

Host.propTypes = {
  socket: PropTypes.shape({
    emit: PropTypes.func.isRequired,
  }).isRequired,
  history: PropTypes.shape({}).isRequired,
  rooms: PropTypes.arrayOf(PropTypes.string).isRequired,
};

const mapStateToProps = ({ rooms }) => ({
  rooms,
});

export default connect(mapStateToProps)(Host);
