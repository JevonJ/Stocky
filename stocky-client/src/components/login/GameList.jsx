import React, { Component } from 'react';

import { ListGroup, ListGroupItem } from 'reactstrap';
import { connect } from 'react-redux';
import { Button, Fade, Popover, PopoverBody, PopoverHeader } from 'reactstrap';
import { ButtonDropdown, DropdownToggle, DropdownMenu, DropdownItem } from 'reactstrap';
import './Welcome.css';
import logo from './logo1.png'; 

class GameList extends Component {
  constructor(props) {
    super(props);

    this.toggle = this.toggle.bind(this);
    this.state = {
	 color: 'green',
      dropdownOpen: false
    };
  }

  toggle() {
    this.setState({
      dropdownOpen: !this.state.dropdownOpen
    });
  }

  setColor(color) {
    this.setState({ color }); 
  }

  changeColor(color) {
    this.props.socket.emit('change color', color);
  }

  
  togglePopover() {
  console.log(this.state.color);
    this.setState({
      popoverOpen: !this.state.popoverOpen,
	  color: '#5E33FF',
    });
  }

 
  

  render() {
    return (
      <Fade in tag="div" timeout={500}>
	 
	  <div style={{backgroundImage:'url(' + logo + ')'}} className="d-flex w-100 h-100 p-3 mx-auto flex-column text-center">
           
		  <header className="masthead mb-auto">
            <div className="inner">
              <h3 className="masthead-brand">Stocky v0.1{this.state.color}</h3>
              <nav className="nav nav-masthead justify-content-center">
                <a className="nav-link active" href="#">Home</a>
                <a className="nav-link" href="#">About</a>
              </nav>
            </div>
          </header>

          <main role="main" className="inner cover">
            <h1 className="cover-heading">Game List</h1>
            <p className="lead">Which game would you like to play ?</p>
            <p className="lead">
              <ButtonDropdown isOpen={this.state.dropdownOpen} toggle={this.toggle}>
        <DropdownToggle caret>
          Select Game
        </DropdownToggle>
        <DropdownMenu>
          
          <DropdownItem>Stocky</DropdownItem>
          <DropdownItem>Brilliant Stocker</DropdownItem>
          <DropdownItem>G Stock Market</DropdownItem>
          
        </DropdownMenu>
      </ButtonDropdown>
            </p>
			 <p className="lead">
              <Button outline color="primary" size="lg" id="Popover1" onClick={() => this.togglePopover()}>Join Game</Button>
             
            </p>
            
          </main>

          <footer className="mastfoot mt-auto">
            <div className="inner">
              <p>
                <a href="https://github.com/JevonJ/Stocky">A Stock Simulator Game</a>, by
                <a href="#" id="Popover1" onClick={() => this.togglePopover()}> Team Titans.</a>
                <Popover
                  placement="top"
                  isOpen={this.state.popoverOpen}
                  target="Popover1"
                  toggle={() => this.togglePopover()}
                >
                  <PopoverHeader>Our Team</PopoverHeader>
                  <PopoverBody>
                    <ul>
                      <li>Jevon Jansz - 15208469</li>
                      <li>A.L. Alvis - 15208464</li>
                      <li>I.J.Ginige - 15209285</li>
                      <li>D.K.D.S.B.Gunawardhane - 16211242</li>
                      <li>S.Tharaniya jj - 15209302</li>
                      <li>N.G.Wickramadivakara - 15209308</li>
                    </ul>
                  </PopoverBody>
                </Popover>
				
				
              </p>
            </div>
          </footer>
        </div>
      </Fade>
    );
  }
}



const mapStateToProps = (state) => {
  console.log('redux state:', state);
};

export default connect(mapStateToProps)(GameList);
