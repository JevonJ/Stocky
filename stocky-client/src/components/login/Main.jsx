import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Popover, PopoverBody, PopoverHeader } from 'reactstrap';
import { Route, withRouter, NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

import Welcome from './Welcome';
import Host from './Host';
import GameList from './GameList';
import Lobby from './Lobby';
import Loading from './Loading';

import BGImage from '../../images/welcome.jpeg';
import './Main.css';

const styles = {
  container: {
    backgroundSize: 'cover',
    display: 'flex',
    color: '#fff',
    backgroundImage: `url(${BGImage})`,
    boxShadow: 'inset 0 0 5rem rgba(0, 0, 0, 1)',
    textShadow: '0 .05rem .1rem rgba(0, 0, 0, .5)',
  },
};

class Main extends Component {
  constructor() {
    super();
    this.state = {
      popoverOpen: false,
    };
  }

  shouldComponentUpdate(nextProps) {
    return this.props.isLoading !== nextProps.isLoading;
  }

  togglePopover() {
    this.setState({
      popoverOpen: !this.state.popoverOpen,
    });
  }

  render() {
    const { socket } = this.props;

    return (
      <div style={styles.container}>
        <div className="d-flex w-100 h-100 p-3 mx-auto flex-column text-center loginMain-container">
          <header className="loginMain-masthead mb-auto">
            <div className="inner">
              <h3 className="loginMain-masthead-brand">Stocky</h3>
              <nav className="nav loginMain-nav-masthead justify-content-center">
                <NavLink
                  exact
                  className="loginMain-nav-link"
                  activeClassName="active"
                  href="/login"
                  to="/login"
                >
                  Home
                </NavLink>
                <NavLink
                  exact
                  className="loginMain-nav-link"
                  activeClassName="active"
                  href="login/about"
                  to="login/about"
                >
                  About
                </NavLink>
              </nav>
            </div>
          </header>

          {
            this.props.isLoading ?
              <Loading />
                :
              <div>
                <Route exact path="/login" component={Welcome} />
                <Route
                  exact
                  path="/login/host-game"
                  component={({ history }) => <Host history={history} socket={socket} />}
                />
                <Route
                  exact
                  path="/login/lobby"
                  component={({ history }) => <Lobby history={history} socket={socket} />}
                />
                <Route
                  exact
                  path="/login/game-List"
                  component={({ history }) => <GameList history={history} socket={socket} />}
                />
                <Route exact path="/login/about" component={Welcome} />
              </div>
          }

          <footer className="loginMain-mastfoot mt-auto">
            <div className="inner">
              <p>
                <a className="loginMain-a" href="https://github.com/JevonJ/Stocky">A Stock Market Simulator Game </a>  by
                <button
                  type="button"
                  className="btn btn-link loginMain-btn-link"
                  id="Popover1"
                  onClick={() => this.togglePopover()}
                >
                    Team Titans.
                </button>
                <Popover
                  placement="top"
                  isOpen={this.state.popoverOpen}
                  target="Popover1"
                  toggle={() => this.togglePopover()}
                >
                  <PopoverHeader>Our Team</PopoverHeader>
                  <PopoverBody>
                    <ul>
                      <li>Jevon Jansz - 15208469</li>
                      <li>A.L. Alvis - 15208464</li>
                      <li>I.J.Ginige - 15209285</li>
                      <li>D.K.D.S.B.Gunawardhane - 16211242</li>
                      <li>S.Tharaniya - 15209302</li>
                      <li>N.G.Wickramadivakara - 15209308</li>
                    </ul>
                  </PopoverBody>
                </Popover>
              </p>
            </div>
          </footer>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => (
  {
    rooms: state.rooms,
    players: state.players,
    isLoading: state.common.isLoading,
  }
);

Main.propTypes = {
  socket: PropTypes.shape({}).isRequired,
  isLoading: PropTypes.bool.isRequired,
};

export default withRouter(connect(mapStateToProps)(Main));
