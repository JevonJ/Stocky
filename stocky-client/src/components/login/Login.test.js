import React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import Loading from './Loading';
import Host from './Host';
import Welcome from './Welcome';
import Lobby from './Lobby';
import GameList from './GameList';
import Main from './Main';

const middlewares = [thunk];
const mockStore = configureStore(middlewares)
const initialState = { rooms: [], user: {}, players: [] };
const store = mockStore(initialState);

it('Loading renders correctly', () => {
  const wrapper = shallow(<Loading />);

  expect(wrapper).toMatchSnapshot();
});

it('Host renders correctly', () => {
  const wrapper = shallow(<Host store={store} socket={{ emit: () => {} }} history={{}} />).dive();

  expect(wrapper).toMatchSnapshot();
});

it('Welcome renders correctly', () => {
  const wrapper = shallow(<Welcome />);

  expect(wrapper).toMatchSnapshot();
});

it('Lobby renders correctly', () => {
  const wrapper = shallow(<Lobby store={store}socket={{ emit: () => {} }} history={{ replace: () => {} }} />).dive();

  expect(wrapper).toMatchSnapshot();
});

it('GameList renders correctly', () => {
  const wrapper = shallow(<GameList store={store}/>);

  expect(wrapper).toMatchSnapshot();
});

it('Main renders correctly', () => {
  const wrapper = shallow(<Main />);

  expect(wrapper).toMatchSnapshot();
});