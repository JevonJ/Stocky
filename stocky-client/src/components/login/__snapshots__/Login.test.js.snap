// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameList renders correctly 1`] = `
<GameList
  dispatch={[Function]}
  players={Array []}
  rooms={Array []}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;

exports[`Host renders correctly 1`] = `
<Fade
  appear={true}
  baseClass="fade"
  baseClassActive="show"
  enter={true}
  exit={true}
  in={true}
  mountOnEnter={false}
  onEnter={[Function]}
  onEntered={[Function]}
  onEntering={[Function]}
  onExit={[Function]}
  onExited={[Function]}
  onExiting={[Function]}
  tag="div"
  timeout={500}
  unmountOnExit={false}
>
  <main
    className="inner loginHost-cover"
    role="main"
  >
    <h1
      className="loginHost-cover-heading"
    >
      Host A Game
    </h1>
    <Form
      tag="form"
    >
      <FormGroup
        row={true}
        tag="div"
      >
        <Label
          for="room"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          Game Room Name
        </Label>
        <Input
          autoComplete="off"
          id="room"
          invalid={false}
          name="room"
          onChange={[Function]}
          type="text"
          valid={false}
          value=""
        />
        <FormFeedback
          tag="div"
          valid={true}
        >
          Sweet! that name is available
        </FormFeedback>
        <FormFeedback
          tag="div"
        />
      </FormGroup>
      <FormGroup
        row={true}
        tag="div"
      >
        <Label
          for="username"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          User name
        </Label>
        <Input
          autoComplete="off"
          id="username"
          invalid={false}
          name="username"
          onChange={[Function]}
          type="text"
          value=""
        />
        <FormFeedback
          tag="div"
        />
      </FormGroup>
      <FormGroup
        row={true}
        tag="div"
      >
        <Label
          for="rounds"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          Number of Rounds
        </Label>
        <Input
          autoComplete="off"
          id="rounds"
          invalid={false}
          max={50}
          min={15}
          name="rounds"
          onChange={[Function]}
          placeholder="Enter number of rounds"
          type="number"
          value={15}
        />
        <FormFeedback
          tag="div"
        />
      </FormGroup>
      <FormGroup
        row={true}
        tag="div"
      >
        <Label
          for="duration"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          Round duration in seconds
        </Label>
        <Input
          autoComplete="off"
          id="duration"
          invalid={false}
          max={90}
          min={30}
          name="duration"
          onChange={[Function]}
          placeholder="Enter round duration (seconds)"
          type="number"
          value={30}
        />
        <FormFeedback
          tag="div"
        />
      </FormGroup>
      <FormGroup
        row={true}
        tag="div"
      >
        <CustomInput
          id="isPrivate"
          label="Private Game"
          name="isPrivate"
          onChange={[Function]}
          type="checkbox"
          value={false}
        />
      </FormGroup>
    </Form>
    <p
      className="lead"
    >
      <Button
        color="success"
        onClick={[Function]}
        outline={true}
        size="lg"
        tag="button"
      >
        Create Game
      </Button>
      <Button
        color="secondary"
        onClick={[Function]}
        outline={true}
        size="lg"
        tag="button"
      >
        Cancel
      </Button>
    </p>
  </main>
</Fade>
`;

exports[`Loading renders correctly 1`] = `
<div
  style={
    Object {
      "paddingLeft": "46%",
    }
  }
>
  <Loading
    color="#fff"
    delay={0}
    height={100}
    type="bars"
    width={100}
  />
</div>
`;

exports[`Lobby renders correctly 1`] = `
<div
  className="animated fadeIn"
>
  <ToastContainer
    autoClose={5000}
    bodyClassName={null}
    className={null}
    closeButton={
      <CloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    draggable={true}
    draggablePercent={80}
    hideProgressBar={true}
    newestOnTop={false}
    pauseOnHover={true}
    pauseOnVisibilityChange={true}
    position="top-right"
    progressClassName={null}
    rtl={false}
    style={null}
    toastClassName={null}
    transition={[Function]}
  />
  <h1
    className="cover-heading"
  >
    Lobby : 
  </h1>
  <p
    className="lead"
  >
    Connected Players
  </p>
  <div
    className="row"
  >
    <div
      className="col col-sm-12 ml-auto"
    >
      <table
        className="table table-hover"
      >
        <thead>
          <tr>
            <th>
              #
            </th>
            <th>
              Username
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col col-sm-12 ml-auto"
    />
  </div>
</div>
`;

exports[`Main renders correctly 1`] = `
<Route
  render={[Function]}
/>
`;

exports[`Welcome renders correctly 1`] = `
<Route
  render={[Function]}
/>
`;
